zenip-42202.md

```diff
@@ -0,0 +1,121 @@
+
+- bugs in the software running the nodes
+
+- connectivity issues
+
+- congestion of blocks (too many transactions and certificates for one single block)
+
+**Note**: The changes proposed in this ZenIP can mitigate such risks without affecting the security of the sidechain.
+
+## Specification
+
+The creation of sidechains v2 is triggered by the consensus rules starting with the activation of the fork point #10.
+
+Any attempt to create a sidechain v2 on mainnet before that height will be rejected.
+
+Sidechains v2 will keep all the features included in v1 in addition to the possibility to enable the non-ceasing behavior by properly setting the `withdrawalEpochLength`.
+
+![Non-ceasable sidechain](ZenIP-42202/ZenIP-42202-1.png)
+
+### Constraints
+
+Sidechains v2 will never cease when setting `withdrawalEpochLength` to “0” at creation time. They are not forced to publish certificates during fixed submission windows. However, they can publish a certificate whenever they want, with the only two following constraints:
+
+- The `epoch` values of certificates must be consecutive (i.e. after a certificate of epoch N the sidechain must publish a certificate of epoch N + 1)
+
+- The `quality` mechanism is disabled, so it’s not possible to publish two or more certificates for the same epoch
+
+These non-ceasable sidechains must satisfy the following constraints:
+
+- **Sidechain Creation**
+
+  - `withdrawalEpochLength` must be set to 0 (to disable the fixed-length epochs and submission windows, otherwise they work exactly as sidechains v1 do).
+
+  - `mainchainBackwardTransferRequestDataLength` must be set to 0 (to disable the MBTR).
+
+  - `wCeasedVk` must be empty (to disable the CSW).
+
+- **Certificates**
+
+  - The `quality` field must always be set to “0”.
+
+  - If mainchain receives first a certificate of epoch N (either in mempool or in a block) and then another certificate of same epoch, the latter is rejected.
+
+  - All certificates must refer (through the `endEpochCumScTxCommTreeRoot` field) to different ordered block. So, for instance, if the mainchain receives a certificate for epoch N referring to the mainchain block M (i.e. setting the `endEpochCumScTxCommTreeRoot `field to the same value specified in the header of the mainchain block M), any certificate coming next must refer to a `endEpochCumScTxCommTreeRoot` (and thus to a block) with higher height than M (M + 1, M + 2, etc., are all good candidates). Such constraint enforces that a certificate of higher epoch refers to a more updated state of the mainchain and of the sidechain.
+
+  - More than one certificate for the same sidechain can be included in the same block (if they respect all the constraints above and are ordered sequentially).
+
+When sidechains v2 are created with `withdrawalEpochLength` not set to 0, they will enable the default ceasing mechanism as it is intended for sidechains v1.
+
+### Maturity
```

Comment:
Because this is a proposal for non-ceasing sidechains, instead of describing the concept of maturity for ceasing sidechains, we could include a link to a page that defines what it means.

Then for the "Maturity" section, I would only keep the last part that is related to non-ceasing sidechains.

Reply:
Unfortunately, we don't have at the moment a document ready to describe maturity and we haven't even agreed about the place to be used for mainchain documentation. I guess that until the decision is made, the only possibility is to briefly describe the maturity concept in the ZenIP itself.

Reply:
That is fine.
I cannot resolve my comments. Feel free to resolve any unclosed ones.